receiver/snmpreceiver/scraper.go

```diff
@@ -39,13 +64,345 @@ func newScraper(logger *zap.Logger, cfg *Config, settings component.ReceiverCrea
 }
 
 // start gets the client ready
-func (s *snmpScraper) start(_ context.Context, _ component.Host) (err error) {
-	return nil
+func (s *snmpScraper) start(_ context.Context, host component.Host) (err error) {
+	s.client, err = newClient(s.cfg, s.logger)
+
+	return err
 }
 
 // scrape collects and creates OTEL metrics from a SNMP environment
 func (s *snmpScraper) scrape(_ context.Context) (pmetric.Metrics, error) {
-	md := pmetric.NewMetrics()
+	if err := s.client.Connect(); err != nil {
+		return pmetric.NewMetrics(), fmt.Errorf("problem connecting to SNMP host: %w", err)
+	}
+	defer s.client.Close()
+
+	// Create the metrics helper which will help manage a lot of the otel metric and resource functionality
+	metricHelper := newOTELMetricHelper(s.settings)
+
+	configHelper := newConfigHelper(s.cfg)
+
+	var scraperErrors scrapererror.ScrapeErrors
+	// Try to scrape scalar OID based metrics
+	s.scrapeScalarMetrics(metricHelper, configHelper, &scraperErrors)
+
+	// Try to scrape column OID based metrics
+	s.scrapeIndexedMetrics(metricHelper, configHelper, &scraperErrors)
+
+	return metricHelper.metrics, scraperErrors.Combine()
+}
+
+// scrapeScalarMetrics retrieves all SNMP data from scalar OIDs and turns the returned scalar data
+// into metrics with optional enum attributes
+func (s *snmpScraper) scrapeScalarMetrics(
+	metricHelper *otelMetricHelper,
+	configHelper *configHelper,
+	scraperErrors *scrapererror.ScrapeErrors,
+) {
+	metricScalarOIDs := configHelper.getMetricScalarOIDs()
+
+	// If no scalar metric configs, nothing else to do
+	if len(metricScalarOIDs) == 0 {
+		return
+	}
+
+	// Retrieve all SNMP data from scalar metric OIDs
+	scalarData := s.client.GetScalarData(metricScalarOIDs, scraperErrors)
+
+	// Create general resource if we're about to make scalar metrics
+	resource := metricHelper.getResource(generalResourceKey)
+	if resource == nil {
+		metricHelper.createResource(generalResourceKey, map[string]string{})
+	}
```

Comment:
Shouldn't we check if nothing was retrieved. This seems like an easy case to detect that we don't need to create the general resource.

Also, once we decide to create it, we don't need to check if it exists, since it couldn't have been created earlier in the scrape.

```suggestion
	if len(scalerData) == 0 {
		return
	}

	metricHelper.createResource(generalResourceKey, map[string]string{})
```

Reply:
@djaglowski i guess that's fair. just because we have scalar oids in the config doesn't mean we'll get data back for them to make the resource